source/1_engine/rex_engine/src/string/stringpool.cpp

```diff
@@ -0,0 +1,155 @@
+#include "rex_engine/string/stringpool.h"
+
+#include "rex_engine/diagnostics/assert.h"
+#include "rex_engine/string/stringentry.h"
+#include "rex_std/functional.h"
+#include "rex_std/internal/utility/pair.h"
+#include "rex_std/unordered_map.h"
+
+namespace rsl
+{
+  inline namespace v1
+  {
+    // https://en.wikipedia.org/wiki/Fowler–Noll–Vo_hash_function
+    rsl::hash_result hash_bytes(const void* data, rsl::count_t size) noexcept
+    {
+      const rsl::hash_result fnv_offset_basis = 2166136261u;
+      const rsl::hash_result fnv_prime        = 16777619u;
+
+      const u8* bytes = static_cast<const u8*>(data);
+
+      rsl::hash_result hash = fnv_offset_basis;
+      for(rsl::count_t i = 0; i < size; ++i)
+      {
+        hash ^= bytes[i];
+        hash *= fnv_prime;
+      }
+
+      return hash;
+    }
+
+    // Template specialization for hash a rsl::string
+    template <>
+    struct hash<rsl::string>
```

Comment:
this already exists in rsl afaik, if not, then this should get added

Reply:
Could you tell me where I can find it?